### ISSUE TITLE
Dovecot PAM authentication regex - lowercase p should be uppercase

### ISSUE THREAD

**Issue by bowaggoner** [fail2ban/fail2ban#2880]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): **v0.10.2**
- OS, including release name/version: **Debian 10 Buster.**
- [**X**] Fail2Ban installed via OS/distribution mechanisms
- [**X**] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
The default dovecot regex for PAM authentication failures does not match the dovecot authentication failure lines in my mail log.
I am referring to line 17 of this file: https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/dovecot.conf
which currently reads

`            ^pam\(\S+,<HOST>(?:,\S*)?\): pam_authenticate\(\) failed: (?:User not known to the underlying authentication module: \d+ Time\(s\)|Authentication failure \(password mismatch\?\)|Permission denied)\s*$`

If I change "password mismatch" to "Password mismatch", it works. (Note also that line 18 contains "Password mismatch".)

Running Dovecot 2.3.4.1 (f79e8e7e4).

#### Steps to reproduce
```
$ cat /var/log/mail.log | grep dovecot | grep pam | tail -n 1 > test.txt

Contents of test.txt, slightly redacted:
Nov 25 18:55:55 MYSERVER dovecot: auth-worker(12182): pam(ozomasi,45.142.120.180): pam_authenticate() failed: Authentication failure (Password mismatch?)

$ fail2ban-regex test.txt /etc/fail2ban/filter.d/dovecot.conf
...
Failregex: 0 total
...

$ sed -i "s/password mismatch/Password mismatch/g" /etc/fail2ban/filter.d/dovecot.conf

$ fail2ban-regex test.txt /etc/fail2ban/filter.d/dovecot.conf
...
Failregex: 1 total
|-  #) [# of hits] regular expression
|   3) [1] ^pam\(\S+,<HOST>(?:,\S*)?\): pam_authenticate\(\) failed: (?:User not known to the underlying authentication module: \d+ Time\(s\)|Authentication failure \(Password mismatch\?\)|Permission denied)\s*$
...
```

#### Any additional information
If I knew this applied to other versions of Dovecot and other systems, I would file a pull request, but I don't know.

Also mentioning that although this seems simple, it took me quite a few hours over several days to discover that this is the reason the regex failed to match! Anyway, thank you and hope this is helpful.

**Comment by sebres** [comment]
> If I knew this applied to other versions of Dovecot and other systems, I would file a pull request, but I don't know.

`[Pp]assword mismatch` is a backwards compatible regex, :)

**Comment by sebres** [comment]
Fixed in 6be1a5a0b1b74ff8c1f3cb07e540f24260a125bf. Thx!
